rfcs/025-tagging_our_resources.md

```diff
@@ -0,0 +1,49 @@
+# RFC 025: Tagging our Terraform resources
+
+**Last updated: 29 May 2020.**
+
+
+
+## Problem statement
+
+We manage our infrastructure in Terraform, but our Terraform configurations are split:
+
+*   Across many repositories
+*   Across multiple directories within each repository
+
+This keeps each configuration small, so it can be planned and applied faster.
+It also creates a hard gap between unrelated services -- for example, a change in the storage service shouldn't affect the catalogue pipeline.
+
+However, this approach can make it hard to find the definition of any particular resource.
+
+If you're looking at a resource in the console, you should be able to tell:
+
+*   Is this resource managed in Terraform?
+*   If so, where is the corresponding Terraform configuration?
+
+
+
+## Solution
+
+We are going to tag resources in a Terraform configuration like so:
+
+```hcl
+tags = {
+  TerraformConfigurationURL = "https://github.com/wellcomecollection/:repo_name/:configuration_path"
```

Comment:
Given configuration might mean the service calling a module - what do we mean here?
If I was provisioning one of our modules, would I expect this to the the provisioner, or the module URL?

If it's the module, might `TerraformRepo` or similar be more explicit?

Reply:
I'm using "configuration" in the [strict Terraform sense](https://www.terraform.io/docs/configuration/index.html#resources-and-modules):

> A **Terraform configuration** consists of a root module, where evaluation begins, along with a tree of child modules created when one module calls another.